data-raw/fontawesome-update.R

```diff
@@ -409,4 +410,52 @@ filenames <- c(
   file.path("webfonts", dir(file.path(source_dir, "webfonts")))
 )
 
+# Copy the complete set of CSS and font files to `inst/fontawesome`
 copy_files(source_dir, dest_dir, filenames)
+
+# Remove font files that won't be supported in this package
+file.remove(
+  "inst/fontawesome/webfonts/fa-brands-400.eot",
+  "inst/fontawesome/webfonts/fa-brands-400.svg",
+  "inst/fontawesome/webfonts/fa-brands-400.ttf",
+  "inst/fontawesome/webfonts/fa-brands-400.woff2",
+  "inst/fontawesome/webfonts/fa-regular-400.eot",
+  "inst/fontawesome/webfonts/fa-regular-400.svg",
+  "inst/fontawesome/webfonts/fa-regular-400.ttf",
+  "inst/fontawesome/webfonts/fa-regular-400.woff2",
+  "inst/fontawesome/webfonts/fa-solid-900.eot",
+  "inst/fontawesome/webfonts/fa-solid-900.svg",
+  "inst/fontawesome/webfonts/fa-solid-900.ttf",
+  "inst/fontawesome/webfonts/fa-solid-900.woff2"
+)
+
+# Patch the `all.css` file to remove entries for all but `woff` icon files
+all_css <-
+  readr::read_file(file = "inst/fontawesome/css/all.css") %>%
+  gsub(
+    "src: url\\(.../webfonts/fa-brands-400.*?\\}",
+    "src: url(\"../webfonts/fa-brands-400.woff\") format(\"woff\"); }",
+    .
+  ) %>%
+  gsub(
+    "src: url\\(.../webfonts/fa-regular-400.*?\\}",
+    "src: url(\"../webfonts/fa-regular-400.woff\") format(\"woff\"); }",
+    .
+  ) %>%
+  gsub(
+    "src: url\\(.../webfonts/fa-solid-900.*?\\}",
+    "src: url(\"../webfonts/fa-solid-900.woff\") format(\"woff\"); }",
+    .
+  )
+readr::write_file(all_css, file = "inst/fontawesome/css/all.css")
+
+# Patch the `all.min.css` file to remove entries for all but `woff` icon files
+all_css_min <-
+  readr::read_file(file = "inst/fontawesome/css/all.min.css") %>%
+  gsub(
+    "@font-face.*$",
+    "@font-face{font-family:\"Font Awesome 5 Brands\";font-style:normal;font-weight:400;font-display:block;url(../webfonts/fa-brands-400.woff) format(\"woff\")}.fab{font-family:\"Font Awesome 5 Brands\"}@font-face{font-family:\"Font Awesome 5 Free\";font-style:normal;font-weight:400;font-display:block;url(../webfonts/fa-regular-400.woff) format(\"woff\")}.fab,.far{font-weight:400}@font-face{font-family:\"Font Awesome 5 Free\";font-style:normal;font-weight:900;font-display:block;url(../webfonts/fa-solid-900.woff) format(\"woff\")}.fa,.far,.fas{font-family:\"Font Awesome 5 Free\"}.fa,.fas{font-weight:900}",
```

Comment:
I feel uneasy about hard-coding this output here, in case they make some changes to this part of the CSS. Can you do the changes with a more intelligent `gsub()`?

Reply:
What I did was adapt the same `gsub()` statements that were used for the `all.css` patching. Works much better now (checked for compliance at https://www.cleancss.com/css-beautify/ and Shiny app is fully functional).